elasticdl/python/common/model_utils.py

```diff
@@ -181,3 +183,79 @@ def get_non_embedding_trainable_vars(model, embedding_layers):
         if not is_embedding_item:
             non_embedding_trainable_vars.append(var)
     return non_embedding_trainable_vars
+
+
+def restore_model_params_from_checkpoint(
+    checkpoint_dir, shard_index, shard_num
```

Comment:
Not sure why we need `shard_index` and `shard_num` here. It seems that this function should load all shards to restore a complete model.

Reply:
We need `shard_index` and `shard_num` when we restore parameters for a PS instance. I will implement in the next PR.

Reply:
I see. The checkpoint is used in two cases:

- export to standard Keras model, usually for serving. In this case, we assume the model size is not very large, and just set ps_num to 1.
- load to PS instance, usually for retraining. The PS instances number could also get changed.

Reply:
Yes!